storm-server/src/test/java/org/apache/storm/scheduler/resource/TestResourceAwareScheduler.java

```diff
@@ -1067,6 +1068,208 @@ public void minCpuWorkerSplitFails() {
+
+        // Get first and last block times for multiple runs and strategies
+        long startTime = Time.currentTimeMillis();
+        for (int strategyIdx = 0; strategyIdx < numStrategies; ++strategyIdx) {
+            String strategy = strategies[strategyIdx];
+
+            for (int run = 0; run < numRuns; ++run) {
+                testResults[strategyIdx][run] = testLargeClusterSchedulingTiming(numNodes, strategy);
+            }
+        }
+
+        // Log median ratios for different strategies
+        LOG.info("TestLargeFragmentedClusterScheduling took {} ms", Time.currentTimeMillis() - startTime);
+        for (int strategyIdx = 0; strategyIdx < numStrategies; ++strategyIdx) {
+            double medianFirstBlockTime = getMedianBlockTime(testResults[strategyIdx], true);
+            double medianLastBlockTime = getMedianBlockTime(testResults[strategyIdx], false);
+            double ratio = medianLastBlockTime / medianFirstBlockTime;
+            LOG.info("{}, FirstBlock {}, LastBlock {} ratio {}", strategies[strategyIdx], medianFirstBlockTime, medianLastBlockTime, ratio);
+        }
+
+        // Check last block scheduling time does not get significantly slower
+        final double[] acceptedStrategyTimeRatios = {6.96, 7.78, 7.75};
+        for (int strategyIdx = 0; strategyIdx < numStrategies; ++strategyIdx) {
+            double medianFirstBlockTime = getMedianBlockTime(testResults[strategyIdx], true);
+            double medianLastBlockTime = getMedianBlockTime(testResults[strategyIdx], false);
+            double ratio = medianLastBlockTime / medianFirstBlockTime;
+
+            double slowSchedulingThreshold = 1.5;
+            assert(ratio < slowSchedulingThreshold * acceptedStrategyTimeRatios[strategyIdx]);
+        }
+    }
+
+    // Create multiple copies of a test topology
+    private void addTopologyBlockToMap(Map<String, TopologyDetails> topologyMap, String baseName, Config config,
+                                       double spoutMemoryLoad, int[] blockIndices) {
+        TopologyBuilder builder = new TopologyBuilder(); // a topology with multiple spouts
+        builder.setSpout("testSpout", new TestSpout(), 1).setMemoryLoad(spoutMemoryLoad);
+        StormTopology stormTopology = builder.createTopology();
+        Map<ExecutorDetails, String> executorMap = genExecsAndComps(stormTopology);
+
+        for (int i = blockIndices[0]; i <= blockIndices[1]; ++i) {
+            TopologyDetails topo = new TopologyDetails(baseName + i, config, stormTopology, 0, executorMap, 0, "user");
+            topologyMap.put(topo.getId(), topo);
+        }
+    }
+
+    /*
+     * Test time to schedule large cluster scheduling with fragmentation
+     */
+    private TimeBlockResult testLargeClusterSchedulingTiming(int numNodes, String Strategy) {
+        Config config = null;
+        if (Strategy.equals(DefaultResourceAwareStrategy.class.getName())) {
+            config = createClusterConfig(10, 10, 0, null);
```

Comment:
I don't know how this would be structured. I can imagine an interface class and then 3 derived classes, each with its own createClusterConfig method, but it sounds like you have something different in mind?

As a recent convert from C++, I think of using function pointers, but that concept is not in Java.

Reply:
No, I meant make the Config a parameter of this method, so `testLargeClusterSchedulingTiming(int numNodes, Config config)`. You can then either do `Map<String, Config> strategyToConfig`, or have a static inner class containing the strategy name and Config in line https://github.com/apache/storm/pull/3092/files#diff-36af6e89e1390a52764241458869fd76R1142.

Not that I think you should use them here, but Java does have something similar to function pointers (not as laissez faire though): https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html